### PR TITLE
feat: allow fetching other datasets (obf, opff, opf)

### LINKED ISSUE #220: Extend ProductDataset to other flavors (obf, opf, opff)
### Story

In the code ([here](https://github.com/openfoodfacts/openfoodfacts-python/blob/6a26850f1ebf85314b82bb31f84d5cafeb6305f2/openfoodfacts/dataset.py#L48)) the `ProductDataset` is hardcoded to fetch only the OFF database.

It would be great to allow choosing other flavors (OBF, OPF, OPFF) as a new parameter.

Btw is there dumps of these datasets or do they behave differently than OFF ?

### PR DESCRIPTION
**Pull request by raphodn**
## Description

The current `ProductDataset` implementation only allows to fetch the OFF (food) data

## Solution

- extend `ProductDataset` to allow fetching other datasets: OBF, OPFF, OPF
- update documentation

## Related issue(s)

- Fixes #220

### CODE REVIEW

File: openfoodfacts/dataset.py [hunk 1]
```diff
@@ -17,12 +17,27 @@
...
 DATASET_FILE_NAMES = {
-    DatasetType.jsonl: "openfoodfacts-products.jsonl.gz",
-    DatasetType.csv: "en.openfoodfacts.org.products.csv.gz",
+    Flavor.off: {
+        DatasetType.jsonl: "openfoodfacts-products.jsonl.gz",
+        DatasetType.csv: "en.openfoodfacts.org.products.csv.gz",
+    },
+    Flavor.obf: {
+        DatasetType.jsonl: "openbeautyfacts-products.jsonl.gz",
+        DatasetType.csv: "en.openbeautyfacts.org.products.csv",
```
Comment (raphodn) [inline openfoodfacts/dataset.py:26]: the obf, opff & opf datasets don't seem to have `.csv.gz` files (getting 404).
so I used plain `.csv` instead

File: openfoodfacts/dataset.py [hunk 2]
```diff
@@ -57,7 +73,12 @@ def get_dataset(
 
 
 class ProductDataset:
-    def __init__(self, dataset_type: DatasetType = DatasetType.jsonl, **kwargs):
+    def __init__(
```
Comment (raphodn) [inline openfoodfacts/dataset.py:76]: slight breaking change, as the documentation gives an explicit usage example with `ProductDataset("csv")`, which will not work anymore

```
// instead
ProductDataset(dataset_type="csv")

// or
from openfoodfacts.types import Flavor
ProductDataset(flavor=Flavor.off, dataset_type="csv")
```

**Comment by raphael0202** [comment]
~The thing is, to my knowledge, we don't have JSONL export of anything else than OFF..
Ex: https://static.openproductsfacts.org/openproductsfacts-products.jsonl.gz returns HTTP 404~

My bad, it's now functional!

File: openfoodfacts/dataset.py (hunk 1 above)
Comment (raphael0202) [reply openfoodfacts/dataset.py:26]: obf, opff, opf use an old version of Product Opener so this is expected. The migration to a more recent version is in progress (but it needs quite a lot of work).

**Review by raphael0202 — APPROVED** [review]
It looks good to me, thank you for the PR!

**Pull request merged**